Vungle/src/main/java/com/mopub/mobileads/VungleInterstitial.java

```diff
@@ -288,69 +282,88 @@ public void run() {
+
+                    final MoPubErrorCode errorCode = mapErrorCode(exception.getExceptionCode());
+
+                    mHandler.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            if (mLoadListener != null) {
+                                mLoadListener.onAdLoadFailed(errorCode);
                             }
-                        });
-                    } else {
-                        MoPubLog.log(getAdNetworkId(), CUSTOM, ADAPTER_NAME,
-                                "interstitial ad is not loaded - Placement ID: " + placementReferenceId);
-                        mHandler.post(new Runnable() {
-                            @Override
-                            public void run() {
-                                if (mLoadListener != null) {
-                                    mLoadListener.onAdLoadFailed(MoPubErrorCode.NETWORK_NO_FILL);
-                                }
-
-                                MoPubLog.log(getAdNetworkId(), LOAD_FAILED, ADAPTER_NAME,
-                                        MoPubErrorCode.NETWORK_NO_FILL.getIntCode(),
-                                        MoPubErrorCode.NETWORK_NO_FILL);
+
+                            MoPubLog.log(getAdNetworkId(), LOAD_FAILED, ADAPTER_NAME, errorCode.getIntCode(), errorCode);
+                        }
+                    });
+                }
+            }
+        }
+
+        @Override
+        public void onAdLoaded(@NonNull String placementId) {
+            if (mPlacementId.equals(placementId)) {
+                if (!mIsPlaying) {
+                    MoPubLog.log(getAdNetworkId(), CUSTOM, ADAPTER_NAME,
+                            "interstitial ad successfully loaded - Placement ID: " + placementId);
+                    mHandler.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            if (mLoadListener != null) {
+                                mLoadListener.onAdLoaded();
                             }
-                        });
-                    }
+                            MoPubLog.log(getAdNetworkId(), LOAD_SUCCESS, ADAPTER_NAME);
+                        }
+                    });
                 }
             }
         }
+
     }
+
+    @NonNull
+    private static MoPubErrorCode mapErrorCode(int error) {
+        switch (error) {
+            case NETWORK_ERROR:
+                return NO_CONNECTION;
+            case PLACEMENT_NOT_FOUND:
+                return MISSING_AD_UNIT_ID;
+            case NO_SERVE:
+                return NETWORK_NO_FILL;
+            case AD_FAILED_TO_DOWNLOAD:
+            case ASSET_DOWNLOAD_ERROR:
+                return VIDEO_DOWNLOAD_ERROR;
+            case AD_UNABLE_TO_PLAY:
+                return VIDEO_PLAYBACK_ERROR;
+            default:
+                return UNSPECIFIED;
+        }
+    }
+
```

Comment:
it duplicates code in VungleBanner. maybe we should either make one of them package-visible and use from both classes and optionally move to some third class in package(leaving package visibility)